### DISCUSSION TITLE
Dnsmasq for UEFI HTTP Boot?

### DISCUSSION THREAD

**Discussion by wf2020** [ipxe/ipxe#569] [Q&A] [answered]
Is it possible to use Dnsmasq for direct http booting?

According to dnsmasq.log the client is:
vendor class: HTTPClient:Arch:00016:UNDI:003016

**Comment by GethDeeo** [comment, marked as answer]
Excerpt from my (working) configuration for UEFI amd64:

```
dhcp-match=set:X86-64_EFI_HTTP,option:client-arch,16
dhcp-userclass=set:iPXE,iPXE
dhcp-option=lan,tag:X86-64_EFI_HTTP,tag:!iPXE,option:bootfile-name,<http url pointing to (bin-x86_64-efi/)ipxe.efi binary>
dhcp-option=lan,tag:X86-64_EFI_HTTP,tag:!iPXE,option:vendor-class,HTTPClient
```
Note: iPXE sets userclass to `iPXE` by default (line 2), hence the `!` in the options.

```
# dnsmasq --version
Dnsmasq version 2.85  Copyright (c) 2000-2021 Simon Kelley
Compile time options: IPv6 GNU-getopt no-DBus UBus no-i18n no-IDN DHCP DHCPv6 no-Lua TFTP conntrack ipset auth cryptohash DNSSEC no-ID loop-detect inotify dumpfile
```

Hope it helps!

**Reply by wf2020** [reply]
Thank you.

Unfortunately I couldn't make dnsmask sending anything. Using ubuntu and dnsmasq version 2.80.

As far as I understood the tag X86-64_EFI_HTTP should be set. So I also tried it with my way of settings:

`dhcp-boot=tag:X86-64_EFI_HTTP,http://192.168.xxx.xxx/ipxe.efi`

**Reply by NiKiZe** [reply]
You can always use tcpdump to see what is on the wire.

**Reply by wf2020** [reply]
Chainloading using snponly.efi now works perfectly for me so direct HTTP booting from a certain NIC is not really necassary for me anymore.

**Comment by Race666** [comment]
Had the same problem that dnsmasq does not respond to UEFI HTTP boot requests. My solution was to set 
```dhcp-pxe-vendor=PXEClient,HTTPClient:Arch:00016```
this is needed if the PXE Client is not standard conform and does not send "PXEClient" in the vendor class field.

**Reply by wf2020** [reply]
I can't find any documentation about dhcp-pxe-vendor. Adding this line (Dnsmasq Version 2.80 at Ubuntu 20.04.4) causes an error.

**Comment by Race666** [comment]
I'm on debian 11, dnsmasq 2.85

https://thekelleys.org.uk/dnsmasq/docs/dnsmasq-man.html

Am 09.06.2022 14:19, schrieb wf2020:

> I can't find any documentation about dhcp-pxe-vendor. Adding this line 
> (Dnsmasq Version 2.80 at Ubuntu 20.04.4) causes an error.
> 
> --
> Reply to this email directly, view it on GitHub [1], or unsubscribe 
> [2].
> You are receiving this because you commented.Message ID: 
> ***@***.***>


Links:
------
[1] 
https://github.com/ipxe/ipxe/discussions/569#discussioncomment-2913169
[2]
